### DISCUSSION TITLE
option to not inject `injected-web.js` if all running userscripts are `// @inject-into content` or there are no running userscripts

### DISCUSSION THREAD

**Discussion by xBZZZZ** [violentmonkey/violentmonkey#1441] [Ideas]
will prevent web page from messing up `injected-web.js` code

**Comment by tophf** [comment]
This is already partially implemented but only when synchronous injection is enabled so indeed, now that our globals are vault-protected, we can move `injectPageSandbox` call into `injectScripts`.

There's no need for an option though, it will happen automatically, because reading an option is asynchronous so it can take an unpredictable amount of time, in other words it may occur after GetInjected message.
